### PR TITLE
Demystifying sql

### PR DESCRIPTION
**Pull request by pm0kjp**
Hi all!  @drelliche, @leemc-data-ed, @rosemm, @franzenr!

This is a **small**, 30-40 minute tops demystifying SQL module.  IDK if it really merits inclusion or not  and my feelings will NOT be hurt if we decide this is kind of orthogonal to what we've been doing so far and doesn't make sense to include.  

Take a gander at it.  NO images so it should be a super quick QA.  I don't have the bandwidth to make major changes so my ask is that you weigh in on whether it should be included with any minor sprucing up or if we should punt until we can get something more meaty.

Feel free to chat in comments, or self-select as a reviewer, depending on who has time!

### CODE REVIEW

**Comment by leemc-data-ed** [comment]
I like it, and I think it would fit well with our on ramp. And I agree, QA shouldn't take very long.
My vote is yes, and I'm happy to QA this one.

File: demystifying_sql/demystifying_sql.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+Which of these correctly describe relational databases?  Choose all the correct options.
+
+[[ ]] Relational databases are an alternative to SQL
+[[X]] Relational databases organize data into tables
+[[ ]] Tables in relational databases are organized into ranks and files
+[[X]] Relational databases permit the extraction of highly specific data
+[[X]] The "relational" in "relational database" refers to data that occurs in multiple places and allows for linking, or "relating" data
+*********
+
+Relational databases are a data storage solution and they store data in tables, which are in turn organized into rows and columns.  Relational databases use SQL as a way to access data, search for data meeting particular conditions, and connect related data, using columns that tables have in common.  This allows for extracting data in very specific and customized way.
```
Comment (leemc-data-ed) [inline demystifying_sql/demystifying_sql.md:140]: There is a class for answer explanations (it's "answer" I think), which I think will make the explanations stand out a bit more.

**Review by leemc-data-ed — CHANGES_REQUESTED** [review]
I think the only thing I noticed is the way the quiz answer explanations appear, in other modules we have them in an "answer" box (using the answer class). But otherwise I think this looks great!

File: demystifying_sql/demystifying_sql.md (hunk 1 above)
Comment (pm0kjp) [reply demystifying_sql/demystifying_sql.md:143]: good call, added

File: demystifying_sql/demystifying_sql.md [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+### When SQL Isn't the Right Tool
+
+Although SQL is a great tool for organizing your data into meaningful datasets for extraction, it is usually not a great tool to use for your actual analysis work.
+
+Despite having many functions for simple text parsing, SQL is not the tool/language you want to use for advanced NLP (Natural Language Processing) work.  Similarly, SQL has some basic statistical functions, but isn't intended to provide statistical analysis at the level of a statistical programming language like R.
+
+SQL also doesn't have any capabilities to directly support data visualization work.
+
+For all of these "downstream analytics" use cases, you will want to use an actual analytical programing language or tool like R or Python.
```
Comment (leemc-data-ed) [inline demystifying_sql/demystifying_sql.md:174]: Did you want "programming" here or "programing"? It's spelled the first way in the overview.

File: demystifying_sql/demystifying_sql.md [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+
+### SQL Implementations
+
+Although all SQL implementations have a similar structure, and the same basic syntax, each different SQL database product often has its own minor variations in dialect.
+
+Colloquially people often refer to the different SQL dialects as different "flavors" of SQL.
+
+Some Popular "Flavors" of SQL:
```
Comment (leemc-data-ed) [inline demystifying_sql/demystifying_sql.md:182]: I'm not sure "Popular" or "Flavors" need to be capitalized here

File: demystifying_sql/demystifying_sql.md [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+
+The basic components of a select statement are the `SELECT` and `FROM` keywords. The `FROM` keyword is used to specify the table or tables that hold the data you're interested in, and the `SELECT` keyword is used to provide a list of columns within those table(s) that you would like returned as output.  
+
+<div class = "hint">
+Many people choose write SQL keywords in all capital letters, but that's so that they stand out clearly in SQL code, not because it's a language requirement.  
+</div>
+
+**Select All Columns**
+
+If you would like to return **all** of the columns from the table(s) specified in your SQL query, you can use the `*` wild card character as shown in the example below.  You'll notice that we put a line break between the asterisk and the `FROM` keyword.  Spaces and line breaks, or **whitespace**, doesn't really matter for SQL.  You can run your code together on a single line, or (preferably) add intendation and spaces that help human readers understand your code more easily.
```
Comment (leemc-data-ed) [inline demystifying_sql/demystifying_sql.md:236]: Should it be "Spaces and line breaks, or **whitespace**, don't really matter for SQL."?

**Review by leemc-data-ed — CHANGES_REQUESTED** [review]
Just found a couple of tiny things in my final read-through, but then I think we're set (for real this time)

File: demystifying_sql/demystifying_sql.md (hunk 2 above)
Comment (pm0kjp) [reply demystifying_sql/demystifying_sql.md:174]: ohhhh the typoooos will fix

File: demystifying_sql/demystifying_sql.md (hunk 4 above)
Comment (pm0kjp) [reply demystifying_sql/demystifying_sql.md:236]: yes!  subject verb agreement ftw!

File: demystifying_sql/demystifying_sql.md (hunk 3 above)
Comment (pm0kjp) [reply demystifying_sql/demystifying_sql.md]: agree, fixed

**Review by leemc-data-ed — APPROVED** [review]
Lgtm! Approved!

**Pull request merged**